### PR TITLE
File-like objects support for h5py.File

### PR DESCRIPTION
**Pull request by aparamon**
A small follow-up for https://github.com/h5py/h5py/issues/1006, actually enabling a work-around for https://github.com/h5py/h5py/issues/839.
Only https://github.com/h5py/h5py/commit/875e3729fe1b321702977cfda351aff5c740eee6 is of interest, sorry for unclean history :shy:

### CODE REVIEW

**Comment by aparamon** [comment]
Please also merge https://github.com/h5py/h5py/pull/1105/commits/f8da0292b06d2bd9b09e35053e8ea11728784497

File: h5py/t.log [hunk 1]
```diff
@@ -0,0 +1,2 @@
+"setup.py" �� ���� ����७��� ��� ���譥�
```
Comment (tacaswell) [inline h5py/t.log]: This seems like an erroneously committed file?

**Comment by tacaswell** [comment]
@aparamon Do you mind if I clean the history up again?

**Comment by aparamon** [comment]
@tacaswell Yes, please do!
My vcs experience was severely poisoned by svn, so now with git I'm trying to avoid mistakes only with moderate success :-/
Thank you for your patience.

**Comment by tacaswell** [comment]
OK, I understand what happened, you started working from your existing feature branch (which included commits I re-based in the other PR).  I am going to cherry-pick two the two interesting commits (875e372 and f8da029) and force-push to this PR.

In the future I suggest a workflow like:

```
git checkout master  # switch to master branch
git pull origin master # get the latest from h5py/h5py 
git checkout -b NEW_WORK_BRANCH  # creates a new branch to work on
# HACK commit HACK
git push aparamon NEW_WORK_BRANCH # publish your branch to the world
# create PR
```

**Comment by tacaswell** [comment]
Pushed the original branch to my fork as a backup https://github.com/tacaswell/h5py/tree/fileobj_bck

**Pull request merged**